test/gtest/intervals.cpp

```diff
@@ -0,0 +1,31 @@
+#include "support/intervals.h"
+#include "ir/utils.h"
+#include "gtest/gtest.h"
+
+using namespace wasm;
+
+TEST(IntervalsTest, TestNoOverlapFound) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 4, 2});
+  intervals.emplace_back(Interval{4, 8, 2});
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals).size(), 2u);
+}
+
+TEST(IntervalsTest, TestOverlapFound) {
```

Comment:
It would be good to add tests for different kinds of overlaps, different input orders, different weights. There are a lot of interesting combinations!

Reply:
done